### PR TITLE
Add ExtendedTestRule

### PR DESCRIPTION
**Pull request by mathieucarbou**

### CODE REVIEW

File: test-tools/src/main/java/org/terracotta/org/junit/rules/ExternalResource.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package org.terracotta.org.junit.rules;
```
Comment (cljohnso) [inline test-tools/src/main/java/org/terracotta/org/junit/rules/ExternalResource.java:1]: The root content of this class was lifted from JUnit but lacks attribution in the README.md for this module.  The unit tests associated with `ExternalResource` should also have been picked up.

Comment (cljohnso) [reply test-tools/src/main/java/org/terracotta/org/junit/rules/ExternalResource.java:1]: Looking at this further, I suggest picking up the version from JUnit 4.13 -- it seems to have addressed the problem.

**Comment by mathieucarbou** [comment]
PR updated. Linux builds won't pass until #7 is merged (mvnw script was not executable).

**Comment by mathieucarbou** [comment]
FYI I have modified the PR to not re-use the JUnit one, that is not able to pass the method description  to after() and before(), and is also not able to throw an exception from after().

So this is a different one now.

`ExtendedTestRule` is currently copied in tc-platform in or `common-test-utilities` module and is used across all our rules.

Also, `ExtendedTestRule` with the description passed to before() is required to correctly replace the superclass of Galvan `BasicExternalCluster` class, which need to get the test description, and to avoid replacing a test failure by a galvan cleanup failure.

Anyway... All of that is cleaned in tc-platform. But it would be nice to have a common behavior.

**Comment by mathieucarbou** [comment]
Note: this class is now copied in Angela, tc-platform and used for galvan.

**Pull request closed without merging**